puzzles/models.py

```diff
@@ -142,6 +158,64 @@ def clear_answer(self, guess):
         self.save()
 
 
+# Hooks for syncing metas and tags
+
+def update_tags_pre_save(sender, instance, **kwargs):
```

Comment:
What is sender? What is instance? 

It seems instance is a puzzle object -- can we rename to "puzzle" for more clarity?

Reply:
sender is the Puzzle class and instance is the instance of the Puzzle that just saved.  I used the variable names from the documentation https://docs.djangoproject.com/en/2.2/ref/signals, but I'll change it to make it more clear.